xbmc/settings/SettingsBase.h

```diff
@@ -103,9 +103,13 @@ class CSettingsBase
    \brief Gets the setting with the given identifier.
 
    \param id Setting identifier
+   \param[in] presenceValidated If is set to false, error messages are ignored because the given
```

Comment:
Why is this needed? shouldn't it just log the error if the setting does not exist?

Reply:
I suggested this change.  There are two instances settings that shouldn't be in the source instance-settings.xml  that were logged as debug message.  The error is when they are included.

Reply:
Can you provide a more detailed example or explain it better? I am not really aware of the instance stuff. Regardless, the bool is pretty much a bad smell

Reply:
I did earlier https://github.com/xbmc/xbmc/pull/23648#issuecomment-1690799635 These two values are not supposed to be in instance-setting.xml as they are added by core.   The bool is basically saying don't log a validation message

Reply:
Then I'd say the proper way of handling this is to avoid logging in case they are built-in instance settings (the case of those two settings). Create some set with the known ignored values and check against it before logging.